client/src/Visit.tsx

```diff
@@ -105,20 +143,31 @@ export const Visit = (): JSX.Element => {
             height: '100%'
           }}
         >
-          <MeetingExperience
-            userId={token.user}
-            token={token.token}
-            displayName="Virtual Appointments User"
-            endpointUrl={config.communicationEndpoint}
-            locator={locator}
-            fluentTheme={config.theme}
-            logoUrl={config.logoUrl}
-            waitingTitle={config.waitingTitle}
-            waitingSubtitle={config.waitingSubtitle}
-            chatEnabled={config.chatEnabled}
-            postCall={config.postCall}
-            onDisplayError={(error) => setError(error)}
-          />
+          {meetingLinkLocator && locator && (
```

Comment:
I think we can add another layer of abstraction to minimize the number of if/else to check what  tyoe if meeting it is. 

First determine if it is a Teams Meeting, Rooms Meeting, or neither. 

Then, if Teams Meeting, render (a new) component which does all the API calls needed for the Teams meeting scenario (fetchConfig, fetchToken, etc.), then inside it render the MeetingExperience for Teams

If Rooms Meeting, render (a new) component which does all the API calls needed for the Rooms meeting scenario (fetchConfig, call the test appointment APIs, etc.), then inside it render the MeetingExperience for Rooms

Else, render the JoinTeamsMeeting component.

This way we only need to check once what type of meeting it is, rather than many times in various places.